contributors/design-proposals/service_links.md

```diff
@@ -0,0 +1,237 @@
+# Explicit service links
+
+## Abstract
+
+Proposes an API for explicit links to services in the pod spec.
+
+## Motivation
+
+Currently, all pods are automatically injected with docker-link-style
+environment variables for each service in their namespaces.  Making service
+dependencies explicit in the PodSpec has a number of advantages:
+
+1.  Reduce accidental coupling
+2.  Reduce the number of IP tables rules that have to be created by the
+    kube-proxy
+3.  Identify relationships between pods and services, allowing user interfaces
+    to display additional information for end users.
+
+## Constraints and Assumptions
+
+1.  The old behavior must remain the default in the v1 API
+2.  It must be possible to migrate off the default to using explicit links
+
+## Use Cases
+
+1.  As a user, I want to be able to explicitly define which pods are injected
+    with information about a service so that I can ensure my pods receive
+    information only about services they actually need to use
```

Comment:
I think people will want restriction, but I'm not sure this will be the path to achieving that. Network policy is currently configured via pod labels, for instance.